docs/installation.rst

```diff
@@ -23,10 +23,46 @@ Redis is the recommended storage type for Flask-Session, as it has the most comp
   Flask-Session versions below 1.0.0 (not yet released), use pickle_ as the default serializer, which may have security implications in production if your storage is ever compromised.
 
 
-Direct support
----------------
+Enhanced Installation Options
+-----------------------------
 
-Flask-Session has an increasing number of directly supported storage and client libraries.
+Flask-Session now supports a variety of storage backends directly through optional dependencies. This simplifies the installation process, allowing you to specify the required storage backend at the time of Flask-Session installation itself.
+
+Below is a guide on how to install Flask-Session with support for the desired storage backend:
+
+Available Storage Options
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To install Flask-Session with support for a specific storage backend, use the following command, replacing ``<storage-option>`` with your chosen backend from the list below:
+
+.. code-block:: bash
+
+    pip install Flask-Session[<storage-option>]
+
+Available storage options and their corresponding ``<storage-option>`` values are:
+
+- **Redis**: ``redis``
+- **Memcached**: ``memcached``
+- **MongoDB**: ``mongodb``
+- **SQLAlchemy**: ``sqlalchemy``
+- **FileSystem**: ``filesystem``
+- **CacheLib**: ``cachelib``
+
+For Developers and Contributors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you're developing with Flask-Session or contributing to its codebase, you might want to install all supported backends along with additional development tools. You can do so by specifying the ``dev`` option:
+
+.. code-block:: bash
+
+    pip install Flask-Session[dev]
+
+This will install Flask-Session along with all the optional dependencies listed under the ``dev`` category in ``pyproject.toml``, facilitating a comprehensive development setup.
```

Comment:
I think dev here is misleading as it wont actually install all libraries required for development, eg pytest. There is PEP735 still in the works but for now this is covered by the requirements files and rye-dependencies. If you use [all] it makes much more sense as installing all extras. 

You could then delete all of these lines as they are covered in contributing